File: 1 - BackEnd/Vanilla/PHP/3 - lang/imprimir/imprimir.php
<?php

/* ##########=========================########## */
/* ######===--- Formas de impresion ---===###### */
/* ##########=========================########## */

// Imprime cualquier tipo de dato.
echo "Imprimimos un texto en pantalla con echo.";
echo("Imprimimos un texto en pantalla con echo.");

print "Funciona para imprimir un mensaje.";
print("Funciona para imprimir un mensaje a modo de funcion.");

// La funcion (printf) es la misma que la de C.
printf("
    Hace referencia a un formato especifico, por ejemplo: 
    Este numero %i es un entero.
", 555);

// ------------------------------- //
// ------ Impresion directa ------ //
// ------------------------------- //

// Imprime directamente el (HTML).
if(true){
    ?>
        <p>
            <!-- ... -->
        </p>
    <?php
}

// --------------------- //
// ------ Heredoc ------ //
// --------------------- //

// De esta manera tambien puede hacerse una impresion.

$my_result = funcion(); // Resultado.

echo <<<HTML
    <div>
        <h1>{$my_variable}</h1>
    </div>
    <div>
        <p>{$my_result}</p>
    </div>
HTML;

// Tambien puede asignarse a una variable como si fuera un string.

$cadena = <<<HTML
    <div>
        <h1>{$my_variable}</h1>
    </div>
    <div>
        <p>{$my_result}</p>
    </div>
HTML;

/* ##########=============########## */
/* ######===--- Escapes ---===###### */
/* ##########=============########## */

# Escapando comillas dobles
echo "Estas son comillas dobles escapadas: \"comillas\"";

# Escapando backslash
echo "Estos son backslash escapados \\backslash\\";

# Comillas simples dentro de comillas dobles
echo "Si usas comillas dobles, puedes usar 'comillas simples' sin ningun problema.";

# Comillas dobles dentro de comillas simples
echo 'Si usas comillas simples, puedes usar "comillas dobles" sin ningun problema.';

File: 1 - BackEnd/Vanilla/Python/1 - lang/imprimir/imprimir.py
# ##########=========================##########
# ######===--- Formas de impresión ---===######
# ##########=========================##########

# Imprime cualquier tipo de dato.
print("Imprimimos un texto en pantalla con print.")

# La función format() permite formatear cadenas de manera similar a printf en C.
print("El número es {}.".format(555)) # El número es 555.
# Variables.
mensaje = "Mi nombre es {} y tengo {} años.".format(nombre, edad)

# ------------------------------- #
# ------ Impresión directa ------ #
# ------------------------------- #

# pero puedes utilizar (f-strings) para formatear texto directamente.
my_variable = "Texto"
my_result = "Resultado"

# Imprimir usando (f-strings) (disponible en Python 3.6+).
print(f"""
    <div>
        <h1>{my_variable}</h1>
    </div>
    <div>
        <p>{my_result}</p>
    </div>
""")

# También puedes usar la función (join) para combinar cadenas con saltos de línea.
cadena = "\n".join([
    f"<div>",
    f"    <h1>{my_variable}</h1>",
    f"</div>",
    f"<div>",
    f"    <p>{my_result}</p>",
    f"</div>"
])

# ##########=============##########
# ######===--- Escapes ---===###### 
# ##########=============##########

# Escapando comillas dobles
print("Estas son comillas dobles escapadas: \"comillas\"")

# Escapando backslash
print("Estos son backslashes escapados \\backslash\\")

# Comillas simples dentro de comillas dobles
print("Si usas comillas dobles, puedes usar 'comillas simples' sin ningún problema.")

# Comillas dobles dentro de comillas simples
print('Si usas comillas simples, puedes usar "comillas dobles" sin ningún problema.')
